src/libraries/System.Runtime.InteropServices.RuntimeInformation/src/System/Runtime/InteropServices/RuntimeInformation/RuntimeInformation.Windows.cs

```diff
@@ -42,68 +41,62 @@ public static Architecture OSArchitecture
-                                s_osArch = Architecture.Arm;
-                                break;
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_AMD64:
-                                s_osArch = Architecture.X64;
-                                break;
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_INTEL:
-                                s_osArch = Architecture.X86;
-                                break;
-                        }
+                            Interop.Kernel32.GetNativeSystemInfo(out Interop.Kernel32.SYSTEM_INFO sysInfo);
 
+                            s_osArch = (int)Map((Interop.Kernel32.ProcessorArchitecture)sysInfo.wProcessorArchitecture);
+                        }
                     }
                 }
 
-                Debug.Assert(s_osArch != null);
-                return s_osArch.Value;
+                return (Architecture)s_osArch;
             }
         }
 
         public static Architecture ProcessArchitecture
         {
             get
             {
-                lock (s_processLock)
+                Debug.Assert(sizeof(Architecture) == sizeof(int));
+
+                if (s_processArch == -1)
                 {
-                    if (null == s_processArch)
+                    lock (s_lock)
                     {
-                        Interop.Kernel32.SYSTEM_INFO sysInfo;
-                        Interop.Kernel32.GetSystemInfo(out sysInfo);
-
-                        switch ((Interop.Kernel32.ProcessorArchitecture)sysInfo.wProcessorArchitecture)
+                        if (s_processArch == -1)
                         {
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_ARM64:
-                                s_processArch = Architecture.Arm64;
-                                break;
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_ARM:
-                                s_processArch = Architecture.Arm;
-                                break;
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_AMD64:
-                                s_processArch = Architecture.X64;
-                                break;
-                            case Interop.Kernel32.ProcessorArchitecture.Processor_Architecture_INTEL:
-                                s_processArch = Architecture.X86;
-                                break;
+                            Interop.Kernel32.GetSystemInfo(out Interop.Kernel32.SYSTEM_INFO sysInfo);
+
+                            s_processArch = (int)Map((Interop.Kernel32.ProcessorArchitecture)sysInfo.wProcessorArchitecture);
                         }
                     }
                 }
 
-                Debug.Assert(s_processArch != null);
-                return s_processArch.Value;
+                return (Architecture)s_processArch;
```

Comment:
FWIW, I still think the locks are superfluous, but if you want to keep them, ok.

```C#
public static Architecture ProcessArchitecture
{
    get
    {
        Debug.Assert(sizeof(Architecture) == sizeof(int));

        int arch = s_processArch;
        if (arch == -1)
        {
            Interop.Kernel32.GetSystemInfo(out Interop.Kernel32.SYSTEM_INFO sysInfo);
            s_processArch = arch = (int)Map((Interop.Kernel32.ProcessorArchitecture)sysInfo.wProcessorArchitecture);
        }

        return (Architecture)arch;
    }
}

Reply:
The less locks the better, #done.